### PR TITLE
refactor #8661 - compute resource form partials

### PR DESCRIPTION
**Pull request by unorthodoxgeek**
removing pointless try methods (try is send for everything except nil, for which it returns nil, so there's no point in something like foo && foo.try(:bar)
renaming the "new" variable to "new_host" to remove possible name confusion, as new can be used as a method call to the Class#new method if the current self is a class (not relevant here, but using reserved names for variable names is very bad for the next developer's mental health :smile:

### CODE REVIEW

File: app/views/compute_resources_vms/form/_vmware.html.erb [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-<%= selectable_f f, :cpus, 1..compute_resource.max_cpu_count, { }, :class => "col-md-2", :disabled => !new, :label => _('CPUs') %>
-<%= selectable_f f, :corespersocket, 1..compute_resource.max_cpu_count, { }, :class => "col-md-2", :disabled => !new, :label => _('Cores per socket') %>
-<%= text_f f, :memory_mb, :class => "col-md-2", :disabled => !new, :label => _("Memory (MB)") %>
-<%= selectable_f f, :cluster, compute_resource.clusters, { }, :class => "col-md-2", :disabled => !new, :label => _('Cluster') %>
-<%# selectable_f f, :resource_pool, compute_resource.resource_pools, { }, :class => "col-md-2", :disabled => !new %>
-<%= select_f f, :path, compute_resource.folders, :path, :to_label , {}, { :label => _("Folder"), :class => "col-md-2", :disabled => !new } %>
-<%= select_f f, :guest_id, compute_resource.guest_types, :first, :last, {}, { :label => _("Guest OS"), :class => "col-md-2", :disabled => !new } %>
-<%= select_f f, :hardware_version, compute_resource.vm_hw_versions, :first, :last, {}, { :label => _("Virtual H/W version"), :class => "col-md-2", :disabled => !new } %>
+<% new_host = @host.try(:created_at).nil? %>
```
Comment (dLobatog) [inline app/views/compute_resources_vms/form/_vmware.html.erb]: .try(:new_record)

Comment (unorthodoxgeek) [inline app/views/compute_resources_vms/form/_vmware.html.erb]: true, will fix, just translated the silly ternary operator :)

Comment (unorthodoxgeek) [inline app/views/compute_resources_vms/form/_vmware.html.erb]: fixed.

**Comment by dLobatog** [comment]
Refactors are sorely needed, something I'm thinking of is that if you want to refactor this, I'd consider moving this to a helper (it's used in several places). Also it'd be awesome if you could take a look at other views where we might be trying to check .new_record? by non-standard means, checking the created_at date for instance... 

Something along the lines of this in application_helper.rb

``` ruby
def new?(f)
  if f.object.present?
    f.object.new_record? 
  else
    f.try(:new_record?)
  end
end
```

Is there any other change than 'new' -> 'new_host' ?

**Comment by unorthodoxgeek** [comment]
@eLobato the problem is that there are 5 files, and the logic is somehow different in 2 of them (why? should it be? is this a bug?), so the helper might not work here.
also, I don't like the f.object assumption in your code. method should accept the object itself IMHO, which would make the method calls annoying at times.

**Comment by dLobatog** [comment]
@unorthodoxgeek Right, the fact that the logic is different in these files is (most likely) an error, they all try to do the same thing. Non-saved objects won't have a `created_at` or `id` field.

The method was just a guideline, adapt as needed :smile: . However, it seems to me that if we pass the object directly, why don't we just call `.new_record` directly.

**Comment by dLobatog** [comment]
@unorthodoxgeek ping?

**Comment by unorthodoxgeek** [comment]
sorry @eLobato, got caught up in that rails 4 branch, I forgot about this... will get to it today.

**Comment by unorthodoxgeek** [comment]
@eLobato - done.

**Comment by dLobatog** [comment]
Merged as 826e18e7cf2d97b4cdaed83142e581f3ecd71e27, thanks @unorthodoxgeek ! I changed a bit the commit title/description.

**Pull request closed without merging**